### ISSUE TITLE
Fix S4136 FP: allow grouped interface implementations 

### ISSUE THREAD

**Issue by JohnyWS** [SonarSource/sonar-dotnet#3035]
### Description

When S4136 is active, it collides with the default cleanup of ReSharper, that preferes grouping interface implementations before method overloads.

Could this be included in S4136 as well, so grouping by interface implementation takes precedence over method name?

### Repro steps

Small code stump to replicate the behavior of the code that is reported as a violation to S4136:

```
interface IValidater<T> { bool Validate(T obj); bool CanValidate(T obj); }
class ValidateHelper : IValidater<string>, IValidater<int>
{
  bool Validate(string obj) => false;
  bool CanValidate(string obj) => false;
  
  bool Validate(int obj) => true;
  bool CanValidate(int obj) => true;
}
```

### Expected behavior

S4136 respects the above sorting of method names, by respecting that grouping of methods via the interfaces they implement take precedence over methods with the same name.

### Actual behavior

S4136 reports an error.

### Known workarounds

None that I know of, except disabling the rule.

### Related information

* SonarC# Version
   8.1 (build 13383)
* If running through the Scanner for MSBuild, its version
   4.8.0

**Comment by andrei-epure-sonarsource** [comment]
thanks @JohnyWS . That makes sense.

**Comment by JohnyWS** [comment]
Thanks @andrei-epure-sonarsource. Yeah it does, if you're using implicit interface implementation at least. But since it's a part of the Java spec of the rule, I'm not sure where the change fits?
